### DISCUSSION TITLE
Does EdgedbCLI always attempts to install the lastest server version? Is there a way to use a version that is already installed?

### DISCUSSION THREAD

**Discussion by CristhianMotoche** [geldata/gel#5055] [Q&A]
In the most recent version of EdgeDB server (`2.10+66168b2`) my project cannot be initialized:
```
$ cat edgedb.toml
[edgedb]
server-version = "2.0"

$ edgedb --version
EdgeDB CLI 2.2.6+7eabbf9

$ edgedb project init 
...
Applied m1beqdb5ehfqrixifgjn7iutztzc3hqovk2wqmfm2iw6tmig5cwv3q (00020.edgeql)
edgedb error: InternalServerError: column source_bc6a5538-aee4-11ed-a588-6170d87012e3~1.id does not exist
  Hint: This is most likely a bug in EdgeDB. Please consider opening an issue ticket at https://github.com/edgedb/edgedb/issues/new?template=bug_report.md
  Server traceback:
      edb.errors.InternalServerError: column source_bc6a5538-aee4-11ed-a588-6170d87012e3~1.id does not exist
```

Therefore, I'm trying to use a version that I have already installed because it was working before:
```
$ edgedb server list-versions --installed-only
┌─────────┬─────────────┬───────────┐
│ Channel │ Version     │ Installed │
│ stable  │ 2.6+db69671 │ ✓         │
│ stable  │ 2.9+4fbadd2 │ ✓         │
└─────────┴─────────────┴───────────┘
```
Unfortunately, it keeps trying to download the latest version and fails again 😢: 
```
$ edgedb project init --server-version 2.9
Found `edgedb.toml` in /repo
Initializing project...
Specify the name of EdgeDB instance to use with this project [default: name]:
> name
Checking EdgeDB versions...
┌────────────────────────┬────────────────────────────────────────────────────────┐
│ Project directory      │ /repo             │
│ Project config         │ /repo/edgedb.toml │
│ Schema dir (non-empty) │ /repo/dbschema    │
│ Installation method    │ portable package                                       │
│ Version                │ 2.10+66168b2                                           │
│ Instance name          │ name                                                   │
└────────────────────────┴────────────────────────────────────────────────────────┘
Downloading package...
00:00:01 [==>                 ] 4.52 MiB/34.36 MiB 6.23 MiB/s | ETA: 5s

```
Does EdgeDB cli discards `--server-version 2.9` in order to install always the latest version? Is there a way to force it to use one of the installed versions?

Please, let me know. Thanks in advance!

**Comment by CristhianMotoche** [comment]
FWIW, I also tried to use `--server-version '=2.9'` but I got this error:
```
$ edgedb project init --server-version '=2.9'
error: Invalid value "=2.9" for '--server-version <server_version>': unsupported version format. Examples: `1.15`, `7`, `3.0-rc.1`
```

**Comment by msullivan** [comment]
The current version of the CLI supports doing `edgedb project init --server-version '=2.9'`.
Try running `edgedb cli upgrade`?
If you're running the latest CLI and it isn't working, then something is going wrong and I'm confused.

**Reply by CristhianMotoche** [reply]
Hey @msullivan Thanks! I used the nightly version and that helped me to use `--server-version '=2.9'`:
```
curl --proto '=https' --tlsv1.2 -sSf https://sh.edgedb.com | sh -s -- --nightly
```

**Reply by msullivan** [reply]
Oh, darn, I had thought that the latest stable CLI had it. I will make sure we release a stable CLI that supports it ASAP.

As a tip, you can switch to the nightly with `edgedb cli upgrade --to-nightly`

**Comment by msullivan** [comment]
Also, could you please file a bug about the regression? If you can't or don't want to post your schema and migration scripts publicly, you could email them to me at sully@edgedb.com.

This sort of regression is quite bad, and I will prioritize fixing it and releasing a version 2.11 ASAP.

**Reply by CristhianMotoche** [reply]
Thank you @msullivan Unfortunately, I cannot share the schema or the migrations due to some NDAs. I'll try to replicate the issue in an isolated environment, though and I'll share it with you. Thanks once again.

**Reply by msullivan** [reply]
OK. I am happy to assist you in any way I can; you can reach out to me at that email or on our discord

**Reply by msullivan** [reply]
@CristhianMotoche Looking at the error and knowing what we changed from 2.9 to 2.10, I suspect the error has to do with applying the expression you need to specify in order to change the type of a property/link or to change it from `multi` to `single` or from `optional` to `required`. And it should be in `00021.edgeql` (since `00020.edgeql`) succeeded. Any information you could share about those details would be great, or that advice may be enough for you to produce a reduced case.

**Reply by msullivan** [reply]
Aha, I believe it to be https://github.com/edgedb/edgedb/issues/5060
